clippy_lints/src/undocumented_unsafe_blocks.rs

```diff
@@ -597,32 +599,18 @@ fn span_from_macro_expansion_has_safety_comment(cx: &LateContext<'_>, span: Span
 
 fn get_body_search_span(cx: &LateContext<'_>) -> Option<Span> {
     let body = cx.enclosing_body?;
-    let map = cx.tcx.hir();
-    let mut span = map.body(body).value.span;
-    let mut maybe_global_var = false;
-    for (_, node) in map.parent_iter(body.hir_id) {
-        match node {
-            Node::Expr(e) => span = e.span,
-            Node::Block(_) | Node::Arm(_) | Node::Stmt(_) | Node::LetStmt(_) => (),
-            Node::Item(hir::Item {
-                kind: ItemKind::Const(..) | ItemKind::Static(..),
-                ..
-            }) => maybe_global_var = true,
-            Node::Item(hir::Item {
-                kind: ItemKind::Mod(_),
-                span: item_span,
-                ..
-            }) => {
-                span = *item_span;
-                break;
+    for (_, parent_node) in cx.tcx.hir().parent_iter(body.hir_id) {
+        match parent_node {
+            Node::Crate(mod_) => return Some(mod_.spans.inner_span),
+            node if let Some((span, _)) = span_and_hid_of_item_alike_node(&node)
+                && !is_const_or_static(&node) =>
+            {
+                return Some(span);
             },
-            Node::Crate(mod_) if maybe_global_var => {
-                span = mod_.spans.inner_span;
-            },
-            _ => break,
+            _ => {},
         }
     }
-    Some(span)
+    None
```

Comment:
Is this correct for things in submodules? I think a static in a submodule (separate file) would get the span of `mod foo;` rather than module's inner span